File: packages/typescript/tests/system/select.test.ts
import { describe } from "vitest";
import { baseIt } from "./helpers.ts";

describe("Select", () => {
  const it = baseIt.override("setup", async ({ setup, skip }) => {
    return async (options) => {
      const result = await setup(options);
      const { driverId, model } = result;

      if (driverId === "appium-ios")
        skip("Appium doesn't support select tool yet");
      if (model.provider === "ollama") skip("Poor instruction following");
      if (model.provider === "deepseek" || model.provider === "xai")
        skip("Requires a separate check agent");

      return result;
    };
  });

  it("selects an option", async ({ expect, setup }) => {
    const { al, $ } = await setup();
    await $.navigate("https://the-internet.herokuapp.com/dropdown");

    await al.check("Option 1 is not selected", { assert: expect.assert });
    await expect(al.check("Option 1 is selected")).rejects.toThrow();
    await al.check("Option 2 is not selected", { assert: expect.assert });
    await expect(al.check("Option 2 is selected")).rejects.toThrow();

    await al.do("select 'Option 1'");

    await al.check("Option 1 is selected", { assert: expect.assert });
    await expect(al.check("Option 1 is not selected")).rejects.toThrow();
    await al.check("Option 2 is not selected", { assert: expect.assert });
    await expect(al.check("Option 2 is selected")).rejects.toThrow();
  });
});

File: packages/python/examples/pytest/select_test.py
from os import getenv

import pytest
from pytest import mark, raises

from alumnium import Provider


@mark.xfail(
    getenv("ALUMNIUM_DRIVER", "selenium") == "appium-ios",
    reason="Appium doesn't support select tool yet",
)
def test_select_option(al, navigate):
    model_provider = al.model.provider
    if model_provider == Provider.OLLAMA:
        pytest.xfail("Poor instruction following")
    if model_provider in (Provider.DEEPSEEK, Provider.XAI):
        pytest.xfail(
            "Requires separate check agent as it prefers to follow "
            "retriever instructions (return `value` instead of `statement`)"
        )

    navigate("https://the-internet.herokuapp.com/dropdown")

    al.check("Option 1 is not selected")
    with raises(AssertionError):
        al.check("Option 1 is selected")

    al.check("Option 2 is not selected")
    with raises(AssertionError):
        al.check("Option 2 is selected")

    al.do("select 'Option 1'")

    al.check("Option 1 is selected")
    with raises(AssertionError):
        al.check("Option 1 is not selected")

    al.check("Option 2 is not selected")
    with raises(AssertionError):
        al.check("Option 2 is selected")
